packages/smithy-core/tests/unit/test_retries.py

```diff
@@ -55,48 +60,228 @@ def test_exponential_backoff_strategy(
+
+
+@pytest.mark.parametrize("max_attempts", [2, 3, 10])
+async def test_standard_retry_strategy(max_attempts: int) -> None:
+    strategy = StandardRetryStrategy(max_attempts=max_attempts)
+    error = CallError(is_retry_safe=True)
+    token = await strategy.acquire_initial_retry_token()
+    for _ in range(max_attempts - 1):
+        token = await strategy.refresh_retry_token_for_retry(
+            token_to_renew=token, error=error
+        )
     with pytest.raises(RetryError):
-        strategy.refresh_retry_token_for_retry(token_to_renew=token, error=error)
+        await strategy.refresh_retry_token_for_retry(token_to_renew=token, error=error)
+
+
+async def test_standard_retry_does_not_retry_unclassified() -> None:
+    strategy = StandardRetryStrategy()
+    token = await strategy.acquire_initial_retry_token()
+    with pytest.raises(RetryError):
+        await strategy.refresh_retry_token_for_retry(
+            token_to_renew=token, error=Exception()
+        )
+
+
+async def test_standard_retry_does_not_retry_when_safety_unknown() -> None:
+    strategy = StandardRetryStrategy()
+    error = CallError(is_retry_safe=None)
+    token = await strategy.acquire_initial_retry_token()
+    with pytest.raises(RetryError):
+        await strategy.refresh_retry_token_for_retry(token_to_renew=token, error=error)
+
+
+async def test_standard_retry_does_not_retry_unsafe() -> None:
+    strategy = StandardRetryStrategy()
+    error = CallError(fault="client", is_retry_safe=False)
+    token = await strategy.acquire_initial_retry_token()
+    with pytest.raises(RetryError):
+        await strategy.refresh_retry_token_for_retry(token_to_renew=token, error=error)
+
+
+async def test_standard_retry_after_overrides_backoff() -> None:
+    strategy = StandardRetryStrategy()
+    error = CallError(is_retry_safe=True, retry_after=5.5)
+    token = await strategy.acquire_initial_retry_token()
+    token = await strategy.refresh_retry_token_for_retry(
+        token_to_renew=token, error=error
+    )
+    assert token.retry_delay == 5.5
+
+
+async def test_standard_retry_quota_consumed_accumulates() -> None:
+    strategy = StandardRetryStrategy()
+    error = CallError(is_retry_safe=True)
+    token = await strategy.acquire_initial_retry_token()
+
+    token = await strategy.refresh_retry_token_for_retry(
+        token_to_renew=token, error=error
+    )
+    first_consumed = token.quota_consumed
+    assert first_consumed == StandardRetryQuota.RETRY_COST
+
+    token = await strategy.refresh_retry_token_for_retry(
+        token_to_renew=token, error=error
+    )
+    assert token.quota_consumed == first_consumed + StandardRetryQuota.RETRY_COST
+
+
+async def test_standard_retry_invalid_max_attempts() -> None:
+    with pytest.raises(ValueError):
+        StandardRetryStrategy(max_attempts=0)
+
+    with pytest.raises(ValueError):
+        StandardRetryStrategy(max_attempts=-1)
```

Comment:
Similar note here, parameterizing this gives us a distinct failure for each test case. Then we know what all is broken at once. If the first statement fails and we fix it, we won't know the second is broken until we run the tests again.